FF1Lib/Armor.cs

```diff
@@ -81,6 +92,362 @@ public void ExpandArmor()
+				armorType = SHIRT;
+				armorClass = HEAVY;
+			    } else if (itemId == Item.Ribbon) {
+				armorType = accessories[rng.Between(0, accessories.Count-1)];
+				armorClass = CLOTH;
+			    } else if (requireType.Count > 0) {
+				armorType = requireType[0];
+				armorClass = requireClasses[0];
+				requireType.RemoveAt(0);
+				requireClasses.RemoveAt(0);
+			    } else if (armorType == SHIRT || armorType == BRACELET) {
+				// Don't roll tier 0 shirts or any extra bracelets
+				if (tier == 0 || armorType == BRACELET) {
+				    while (armorType == SHIRT || armorType == BRACELET) {
+					armorType = rng.Between(0, 7);
+				    }
+				}
+			    }
+
+			    var name = "";
+			    byte weight = 0;
+			    byte absorb = 0;
+			    byte elementalResist = 0;
+			    byte spellIndex = 0xFF;
+			    ArmorType type = ArmorType.ARMOR;
+			    ushort permissions = 0;
+
+			    absorb = (byte)RangeScale(armorDefenseBase[armorClass, armorType], .7, 1.4, 1.0, rng);
+			    weight = (byte)RangeScale(armorEvadePenaltyBase[armorClass, armorType], .7, 1.4, 1.0, rng);
+
+			    int nameClass = armorClass;
+			    name = classNames[nameClass][rng.Between(0, classNames[nameClass].Length-1)];
+			    switch (armorClass) {
+				case CLOTH:
+				    permissions = 0xFFF;
+				    break;
+				case LIGHT:
+				    permissions = (ushort)(EquipPermission.Fighter|EquipPermission.Knight|
+							   EquipPermission.Thief|EquipPermission.Ninja|
+							   EquipPermission.BlackBelt|EquipPermission.Master|
+							   EquipPermission.RedMage|EquipPermission.RedWizard);
+				    break;
+				case MEDIUM:
+				    permissions = (ushort)(EquipPermission.Fighter|EquipPermission.Knight|
+							   EquipPermission.Ninja|
+							   EquipPermission.RedMage|EquipPermission.RedWizard);
+				    break;
+				case HEAVY:
+				    permissions = (ushort)(EquipPermission.Fighter|EquipPermission.Knight|
+							   EquipPermission.Ninja);
+				    break;
+				case KNIGHT:
+				    permissions = (ushort)EquipPermission.Knight;
+				    break;
+			    }
+
+			    var spells = new List<FF1Lib.Spell>();
+			    if (itemId == Item.PowerGauntlets) {
+				// cast FAST, SABR or TMPR
+				spells = new List<FF1Lib.Spell>(spellHelper.FindSpells(SpellRoutine.Fast, SpellTargeting.Any).
+								Concat(spellHelper.FindSpells(SpellRoutine.Fast, SpellTargeting.Any)).
+								Select(s => s.Id));
+
+			    } else if (itemId == Item.WhiteShirt) {
+				// cast INV2, FOG2, or WALL
+				spells = new List<FF1Lib.Spell>(spellHelper.FindSpells(SpellRoutine.Ruse, SpellTargeting.AllCharacters).
+								Concat(spellHelper.FindSpells(SpellRoutine.ArmorUp, SpellTargeting.AllCharacters)).
+								Concat(spellHelper.FindSpells(SpellRoutine.DefElement, SpellTargeting.Any, (SpellElement)0xFF)).
+								Select(s => s.Id));
+			    } else if (itemId == Item.BlackShirt) {
+				// Any AOE damage
+				spells = new List<FF1Lib.Spell>(spellHelper.FindSpells(SpellRoutine.Damage, SpellTargeting.AllEnemies).

```

Comment:
Isn't that a bit excessive. It includes Fade and Nuke. Might turn some hardcore people off. Consider restricting this to a low damage, elemental AOE. Additionally you can provide a flag that loosens this up to include any AOE.

Reply:
Now filters out non-elemental AOE damage spells